### PR TITLE
Merged some categories in the report file thanks to their WSTG code

### PR DESCRIPTION
**Pull request by JulienTD**
## Description

In this pull request, I have:
- merged the categories: 'SQL Injection' and 'Blind SQL Injection' into 'SQL Injection'
- separated the category 'Cross Site Scripting' into 'Stored Cross Site Scripting' and 'Reflected Cross Site Scripting'

## Related Issue(s)
 
#47

### CODE REVIEW

File: wapitiCore/definitions/csp.py [hunk 1]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Content Security Policy Configuration")
 SHORT_NAME = _("CSP Configuration")
 
+WSTG_CODE = ["WSTG-CONF-12"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/csp.py:28]: Please update OWASP reference to [https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/12-Test_for_Content_Security_Policy](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/12-Test_for_Content_Security_Policy)

File: wapitiCore/definitions/dangerous_resource.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Potentially dangerous file")
 SHORT_NAME = NAME
 
+WSTG_CODE = ["WSTG-CONF-04"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/dangerous_resource.py:28]: Related to the use of vulnerable compoents : WSTG-CONF-01
https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/01-Test_Network_Infrastructure_Configuration

File: wapitiCore/definitions/exec.py [hunk 3]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Command execution")
 SHORT_NAME = NAME
 
+WSTG_CODE = ["WSTG-INPV-12"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/exec.py:28]: Please update OWASP reference to https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/12-Testing_for_Command_Injection

File: wapitiCore/definitions/file.py [hunk 4]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Path Traversal")
 SHORT_NAME = NAME
 
+WSTG_CODE = ["WSTG-ATHZ-01"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/file.py:28]: Please update OWASP reference to https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/05-Authorization_Testing/01-Testing_Directory_Traversal_File_Include

File: wapitiCore/definitions/htaccess.py [hunk 5]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Htaccess Bypass")
 SHORT_NAME = NAME
 
+WSTG_CODE = ["WSTG-CONF-06"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/htaccess.py:28]: Please add reference to https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods

File: wapitiCore/definitions/http_headers.py [hunk 6]
```diff
@@ -24,6 +24,18 @@
 NAME = _("HTTP Secure Headers")
 SHORT_NAME = NAME
 
+WSTG_CODE_FRAME_OPTIONS = ["WSTG-CLNT-09"]
```
Comment (bretfourbe) [inline wapitiCore/definitions/http_headers.py:27]: Please add reference https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/11-Client-side_Testing/09-Testing_for_Clickjacking

File: wapitiCore/definitions/http_headers.py [hunk 7]
```diff
@@ -24,6 +24,18 @@
...
+WSTG_CODE_XSS_PROTECTION = ["WSTG-INPV-01"]
```
Comment (bretfourbe) [inline wapitiCore/definitions/http_headers.py:28]: + WSTG-INPV-02
https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/01-Testing_for_Reflected_Cross_Site_Scripting
https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/02-Testing_for_Stored_Cross_Site_Scripting

File: wapitiCore/definitions/http_headers.py [hunk 8]
```diff
@@ -24,6 +24,18 @@
...
+WSTG_CODE_CONTENT_TYPE_OPTIONS = ["WSTG-ATHN-01"]
```
Comment (bretfourbe) [inline wapitiCore/definitions/http_headers.py:29]: no idea for this one ?

File: wapitiCore/definitions/http_headers.py [hunk 9]
```diff
@@ -24,6 +24,18 @@
...
+WSTG_CODE_STRICT_TRANSPORT_SECURITY = ["WSTG-CONF-07"]
```
Comment (bretfourbe) [inline wapitiCore/definitions/http_headers.py:30]: https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/07-Test_HTTP_Strict_Transport_Security

File: wapitiCore/definitions/internal_error.py [hunk 10]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Internal Server Error")
 SHORT_NAME = NAME
 
+WSTG_CODE = ["WSTG-ERRH-01"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/internal_error.py:28]: https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/08-Testing_for_Error_Handling/01-Testing_For_Improper_Error_Handling

File: wapitiCore/definitions/redirect.py [hunk 11]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Open Redirect")
 SHORT_NAME = NAME
 
+WSTG_CODE = ["WSTG-CLNT-04"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/redirect.py:28]: https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/11-Client-side_Testing/04-Testing_for_Client-side_URL_Redirect

File: wapitiCore/definitions/reflected_xss.py [hunk 12]
```diff
@@ -25,6 +25,8 @@
 
 SHORT_NAME = _("XSS")
 
+WSTG_CODE = ["WSTG-INPV-01"]
```
Comment (bretfourbe) [inline wapitiCore/definitions/reflected_xss.py:28]: https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/01-Testing_for_Reflected_Cross_Site_Scripting

File: wapitiCore/definitions/resource_consumption.py [hunk 13]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Resource consumption")
 SHORT_NAME = NAME
 
+WSTG_CODE = [""]
```
Comment (bretfourbe) [inline wapitiCore/definitions/resource_consumption.py:27]: WSTG-ERRH-01 ?
https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/08-Testing_for_Error_Handling/01-Testing_For_Improper_Error_Handling

File: wapitiCore/definitions/sql.py [hunk 14]
```diff
@@ -24,6 +24,8 @@
 NAME = _("SQL Injection")
 SHORT_NAME = _("SQLI")
 
+WSTG_CODE = ["WSTG-INPV-05"]
```
Comment (bretfourbe) [inline wapitiCore/definitions/sql.py:27]: https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/05-Testing_for_SQL_Injection

File: wapitiCore/definitions/ssrf.py [hunk 15]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Server Side Request Forgery")
 SHORT_NAME = _("SSRF")
 
+WSTG_CODE = ["WSTG-INPV-19"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/ssrf.py:28]: https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/19-Testing_for_Server-Side_Request_Forgery

File: wapitiCore/definitions/subdomain_takeovers.py [hunk 16]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Subdomain takeover")
 SHORT_NAME = NAME
 
+WSTG_CODE = ["WSTG-CONF-10"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/subdomain_takeovers.py:28]: https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/10-Test_for_Subdomain_Takeover

File: wapitiCore/definitions/timesql.py [hunk 17]
```diff
@@ -24,6 +24,8 @@
 NAME = _("Blind SQL Injection")
 SHORT_NAME = _("Blind SQLI")
 
+WSTG_CODE = ["WSTG-INPV-05"]
```
Comment (bretfourbe) [inline wapitiCore/definitions/timesql.py:27]: https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/05-Testing_for_SQL_Injection

File: wapitiCore/definitions/xss.py [hunk 18]
```diff
@@ -25,6 +25,8 @@
 
 SHORT_NAME = _("XSS")
 
+WSTG_CODE = ["WSTG-INPV-02"]
```
Comment (bretfourbe) [inline wapitiCore/definitions/xss.py:28]: we should keep stored_xxs.py ?

https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/02-Testing_for_Stored_Cross_Site_Scripting

File: wapitiCore/definitions/xxe.py [hunk 19]
```diff
@@ -24,6 +24,8 @@
 NAME = _("XML External Entity")
 SHORT_NAME = _("XXE")
 
+WSTG_CODE = ["WSTG-INPV-07"]
+
```
Comment (bretfourbe) [inline wapitiCore/definitions/xxe.py:28]: https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/07-Input_Validation_Testing/07-Testing_for_XML_Injection

File: wapitiCore/definitions/xss.py (hunk 18 above)
Comment (JulienTD) [reply wapitiCore/definitions/xss.py:28]: Uh, do you want to rename the file ?

Comment (JulienTD) [reply wapitiCore/definitions/xss.py:28]: I tried to rename the xss to stored_xss, but the program doesn't work anymore, it seems that the attack modules are linked to the definitions by their id.

File: wapitiCore/definitions/reflected_xss.py [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
```
Comment (bretfourbe) [inline wapitiCore/definitions/reflected_xss.py:19]: remove this ?

Comment (bretfourbe) [reply wapitiCore/definitions/reflected_xss.py:19]: then looks good to me

Comment (JulienTD) [reply wapitiCore/definitions/reflected_xss.py]: done

**Review by bretfourbe — APPROVED** [review]

**Pull request merged**